kratos/geometries/geometry.h

```diff
@@ -3176,20 +3182,45 @@ class Geometry
 
 
 private:
-    ///@name Static Member Variables
-    ///@{
-
-    // static const GeometryData msEmptyGeometryData;
-
-    ///@}
     ///@name Member Variables
     ///@{
 
+    std::uint64_t mId;
+
     GeometryData const* mpGeometryData;
 
     static const GeometryDimension msGeometryDimension;
 
     PointsArrayType mPoints;
+
+    ///@}
+    ///@name Id Bit-Change Operations
+    ///@{
+
+    /// Checks first bit in Id. 0 -> id; 1 -> name/ string
+    static inline bool IsIdGeneratedFromString(IndexType Id)
+    {
+        return Id & (IndexType(1) << 63);
```

Comment:
You may change this to `return Id & (IndexType(1) << (sizeof(IndexType) - 1));` to be compatible with 32 bit